packages/rocketchat-ui-admin/client/SettingsCachedCollection.js

```diff
@@ -0,0 +1,31 @@
+import _ from 'underscore';
+
+export class PrivateSettingsCachedCollection extends RocketChat.CachedCollection {
+	constructor() {
+		super({
+			name: 'private-settings',
+			eventType: 'onLogged'
+		});
+	}
+
+	setupListener(eventType, eventName) {
```

Comment:
The redefinition is necessary as the change of permissions has to trigger the Settings-collection to be refreshed. We'll have to listen to an event "authorization changed".